src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -472,31 +455,26 @@ private void WriteGreenVisitor(bool withResult)
             OpenBlock();
             foreach (var node in nodes.OfType<Node>())
             {
-                WriteLine("public virtual {0} Visit{1}({2} node) => this.DefaultVisit(node);", withResult ? "TResult" : "void", StripPost(node.Name, "Syntax"), node.Name);
+                WriteLine($"public virtual {(withResult ? "TResult" : "void")} Visit{StripPost(node.Name, "Syntax")}({node.Name} node) => this.DefaultVisit(node);");
             }
             CloseBlock();
         }
 
         private void WriteGreenUpdateMethod(Node node)
         {
             WriteLine();
-            Write("public {0} Update(", node.Name);
-
-            // parameters
-            foreach (var field in node.Fields)
+            Write($"public {node.Name} Update(");
+            Write(CommaJoin(node.Fields.Select(f =>
             {
-                if (field != node.Fields.First())
-                    Write(", ");
-
                 var type =
-                    field.Type == "SyntaxNodeOrTokenList" ? "Microsoft.CodeAnalysis.Syntax.InternalSyntax.SyntaxList<CSharpSyntaxNode>" :
-                    field.Type == "SyntaxTokenList" ? "Microsoft.CodeAnalysis.Syntax.InternalSyntax.SyntaxList<SyntaxToken>" :
-                    IsNodeList(field.Type) ? "Microsoft.CodeAnalysis.Syntax.InternalSyntax." + field.Type :
-                    IsSeparatedNodeList(field.Type) ? "Microsoft.CodeAnalysis.Syntax.InternalSyntax." + field.Type :
-                    field.Type;
-
-                Write("{0} {1}", type, CamelCase(field.Name));
-            }
+                    f.Type == "SyntaxNodeOrTokenList" ? "Microsoft.CodeAnalysis.Syntax.InternalSyntax.SyntaxList<CSharpSyntaxNode>" :
```

Comment:
I feel like I'm getting deja vu, did we just change a complex if-else chain similar to this one to a switch expression? Not that it matters that much here, the goal seems to be just mechanical translation right now.

Reply:
yeah, ther'es a lot of opportunities for cleanup/consistency here.  it's never going to happen in one PR, but rather over a set of many steps :)